pkg/controller/master/upgrade/upgrade_repo.go

```diff
@@ -335,8 +351,24 @@ func (r *Repo) deleteVM() error {
 		}
 	}
 
-	logrus.Infof("Delete upgrade repo VM %s/%s", vm.Namespace, vm.Name)
-	return r.h.vmClient.Delete(vm.Namespace, vm.Name, &metav1.DeleteOptions{})
+	err = r.h.vmClient.Delete(vm.Namespace, vm.Name, &metav1.DeleteOptions{})
+	if err == nil {
+		return nil
+	}
+	if apierrors.IsNotFound(err) {
+		return nil
+	}
+	return fmt.Errorf("failed to delete repo vm %s/%s error %w", vm.Namespace, vm.Name, err)
+}
+
+func (r *Repo) deleteService() error {
+	nm := r.getRepoServiceName()
+	// no serviceCache, delete via client directly
```

Comment:
The cache interface doesn't have a `Delete` method anyways?

Reply:
with a cache, the usual code is: get the object, if it is existing, then delete; 
without a cache, can only blindly delete and check return code to decide whether stop or reconciller